src/main/java/com/google/sps/ChatWebSocket.java

```diff
@@ -59,6 +59,16 @@
      * Verifies if the user is authenticated, if this is not the case the
      * connection is closed.
      */
+
+    public ChatWebSocket() {}
```

Comment:
If these constructors are only needed for the unit test, you can make them package-private instead of public. You can do that by removing the `public` modifier.

Reply:
Would this affect the possibility of instantiating this class from an outside package?

Reply:
As a rule of thumb, we always want to reduce the visibility as much as possible  http://www.thefinestartist.com/effective-java/13